app/helpers/application_helper/button/delete_server.rb

```diff
@@ -0,0 +1,17 @@
+class ApplicationHelper::Button::DeleteServer < ApplicationHelper::Button::ZoneDeleteServer
+  needs :@record
```

Comment:
Class instance variable is not shared between parent and children. Also this line get's executed at class definition time. So imho this is actually needed.